packages/middleware-flexible-checksums/src/getChecksumAlgorithmForRequest.spec.ts

```diff
@@ -1,43 +1,71 @@
+      it("returns undefined if requestChecksumRequired is false", () => {
+        expect(getChecksumAlgorithmForRequest({}, { ...mockOptions, requestChecksumRequired: false })).toBeUndefined();
+      });
     });
-  });
 
-  describe("when requestAlgorithmMember is not set in input", () => {
-    const mockOptions = { requestAlgorithmMember: mockRequestAlgorithmMember };
+    describe(`when requestChecksumCalculation is '${RequestChecksumCalculation.WHEN_SUPPORTED}'`, () => {
+      const mockOptions = { requestChecksumCalculation: RequestChecksumCalculation.WHEN_SUPPORTED };
 
-    it("returns MD5 if requestChecksumRequired is set", () => {
-      expect(getChecksumAlgorithmForRequest({}, { ...mockOptions, requestChecksumRequired: true })).toEqual(
-        ChecksumAlgorithm.MD5
-      );
+      it(`returns ${DEFAULT_CHECKSUM_ALGORITHM} if requestChecksumRequired is set`, () => {
+        expect(getChecksumAlgorithmForRequest({}, { ...mockOptions, requestChecksumRequired: true })).toEqual(
+          DEFAULT_CHECKSUM_ALGORITHM
+        );
+      });
+
+      it(`returns ${DEFAULT_CHECKSUM_ALGORITHM} if requestChecksumRequired is false`, () => {
+        expect(getChecksumAlgorithmForRequest({}, { ...mockOptions, requestChecksumRequired: false })).toEqual(
+          DEFAULT_CHECKSUM_ALGORITHM
+        );
+      });
     });
+  });
 
-    it("returns undefined if requestChecksumRequired is false", () => {
-      expect(getChecksumAlgorithmForRequest({}, { ...mockOptions, requestChecksumRequired: false })).toBeUndefined();
+  describe("returns undefined if input[requestAlgorithmMember] is not set", () => {
+    describe.each([true, false])("when requestChecksumRequired='%s'", (requestChecksumRequired) => {
+      it.each([RequestChecksumCalculation.WHEN_SUPPORTED, RequestChecksumCalculation.WHEN_REQUIRED])(
+        "when requestChecksumCalculation='%s'",
+        (requestChecksumCalculation) => {
+          const mockOptions = {
+            requestChecksumRequired,
+            requestChecksumCalculation,
+            requestAlgorithmMember: mockRequestAlgorithmMember,
+          };
+          expect(getChecksumAlgorithmForRequest({}, mockOptions)).toBeUndefined();
+        }
+      );
     });
   });
 
   it("throws error if input[requestAlgorithmMember] if not supported by client", () => {
     const unsupportedAlgo = "unsupportedAlgo";
     const mockInput = { [mockRequestAlgorithmMember]: unsupportedAlgo };
-    const mockOptions = { requestChecksumRequired: true, requestAlgorithmMember: mockRequestAlgorithmMember };
+    const mockOptions = {
+      requestChecksumRequired: true,
+      requestAlgorithmMember: mockRequestAlgorithmMember,
+      requestChecksumCalculation: RequestChecksumCalculation.WHEN_REQUIRED,
+    };
     expect(() => {
       getChecksumAlgorithmForRequest(mockInput, mockOptions);
-    }).toThrowError(
+    }).toThrow(
```

Comment:
what's the difference between toThrowError and toThrow?

Reply:
They're same as per maintainer of vitest https://github.com/vitest-dev/vitest/discussions/2570#discussioncomment-4503284

Reply:
Changed to specific `toThrowError` in [`b49d70e` (#6750)](https://github.com/aws/aws-sdk-js-v3/pull/6750/commits/b49d70e8c754b20f170d7fe240f1f2773c3c16b6)